include/TestFacility.h

```diff
@@ -35,12 +35,26 @@ namespace ChimeraTK {
       Application::getInstance().initialise();
     }
 
-    /** Start the application. This simply calls Application::run(). Since the
-     * application is in testable mode, it
-     *  will be ??? TODO define precisely what happens on start up */
+    /** Start the application in testable mode. */
     void runApplication() const {
+      // send default values for all control system variables
+      for(auto& pv : pvManager->getAllProcessVariables()) {
+        callForType(pv->getValueType(), [&pv, this](auto arg) {
+          if(!pv->isWriteable()) return;
+          typedef decltype(arg) T;
+          auto pv_casted = boost::dynamic_pointer_cast<NDRegisterAccessor<T>>(pv);
+          auto table = boost::fusion::at_key<T>(defaults.table);
+          if(table.find(pv->getName()) != table.end()) {
+            pv_casted->accessChannel(0) = table.at(pv->getName());
+          }
+          pv_casted->write();
+        });
+      }
+      // start the application
```

Comment:
What is happening if I am not in testable mode? There are tests which have to just use app.run(). And all the real adapters do so as well. Shouldn't this be in app.run()?
I have to admit I don't fully understand what it does. Which side of the pvManager is it (CS or dev)?

Reply:
Real adapters don't use a TestFacility. If you write a test with TestFacility but without testable mode you have to know what you are doing. These things cannot be in app.run() because you might have control over them and hence have disabled the testable mode.

The TestFacility kind of replaces a real adapter, hence the pvManager here is the CS-side manager.